src/Microsoft.IdentityModel.Tokens/Encryption/EcdhKeyExchangeProvider.cs

```diff
@@ -56,51 +52,39 @@ public class EcdhKeyExchangeProvider //todo: rename to KeyExchangeProvider, or E
         /// <summary>
         /// Initializes a new instance of <see cref="EcdhKeyExchangeProvider"/> used for CEKs
         /// <param name="privateKey">The <see cref="ECDsaSecurityKey"/> that will be used for cryptographic operations and represents the private key.</param>
-        /// <param name="publicKey">The <see cref="ECDsaSecurityKey"/> that will be used for cryptographic operations and represents the public key.</param>
+        /// <param name="publicKey">The <see cref="SecurityKey"/> that will be used for cryptographic operations and represents the public key.</param>
         /// <param name="alg">alg header parameter value.</param>
         /// <param name="enc">enc header parameter value.</param>
         /// </summary>
-        public EcdhKeyExchangeProvider(ECDsaSecurityKey privateKey, ECDsaSecurityKey publicKey, string alg, string enc)
+        public EcdhKeyExchangeProvider(ECDsaSecurityKey privateKey, SecurityKey publicKey, string alg, string enc)
```

Comment:
An ECDsaSecurityKey private key could be represented by a JsonWebKey, unlikely since JsonWebKey is normally associated with public metadata.
That said, it is possible so both parameters could be SecurityKey.

Reply:
Changed to SecurityKey and necessary changes below to accommodate to it.